engine/jobmaster/dm/openapi.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dm
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/dm/dm/pb"
+	"github.com/pingcap/tiflow/engine/jobmaster/dm/openapi"
+	dmpkg "github.com/pingcap/tiflow/engine/pkg/dm"
+)
+
+func (jm *JobMaster) initOpenAPI(router *gin.RouterGroup) {
+	wrapper := openapi.ServerInterfaceWrapper{
+		Handler: jm,
+	}
+	// copy from openapi.RegisterHandlersWithOptions
+	router.DELETE("/binlog/tasks/:task-name", wrapper.DMAPIDeleteBinlogOperator)
+
+	router.GET("/binlog/tasks/:task-name", wrapper.DMAPIGetBinlogOperator)
+
+	router.POST("/binlog/tasks/:task-name", wrapper.DMAPISetBinlogOperator)
+
+	router.GET("/config", wrapper.DMAPIGetJobConfig)
+
+	router.PUT("/config", wrapper.DMAPIUpdateJobConfig)
+
+	router.GET("/schema/tasks/:task-name", wrapper.DMAPIGetSchema)
+
+	router.PUT("/schema/tasks/:task-name", wrapper.DMAPISetSchema)
+
+	router.GET("/status", wrapper.DMAPIGetJobStatus)
+
+	router.PUT("/status", wrapper.DMAPIOperateJob)
```

Comment:
should use POST according to out OpenAPI doc

Reply:
It's not create status, it's change status, so I think PUT is more accurate according to the doc.